ex03.exs

```diff
@@ -51,11 +52,42 @@ defmodule Ex03 do
   """
 
   def pmap(collection, process_count, function) do
-    # your code goes here
-    # I'm hoping to see a simple pipeline as the body of this function...
+    collection
+      |> splitChunks(process_count) 
+      |> convertProcess(function) 
+      |> runProc()
   end
 
-  # and here...
+  def splitChunks(collection, process_count) do
+    l = Enum.count(collection)
+    cE = div(l, process_count)
+    Enum.chunk_every(collection, cE)
+  end
+
+  def convertProcess(collection, function) do
+    Enum.map(collection, fn minList -> spawn fn() -> applyFunc(minList, function) end end)
+  end
+
+  def applyFunc(collection, function) do
```

Comment:
The reason it doesn't run in parallel is this function.

What you're doing is spawning n processes. These then wait for you to send them work.

But, when you do send that work, you immediately wait for them to complete. This means that process 1 ones, then process 2, etc.

To get the parallelism, you need to get them all started, and only then start waiting for results to come back.

# -10